atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/backup/CqlCluster.java

```diff
@@ -0,0 +1,142 @@
+ */
+
+package com.palantir.atlasdb.cassandra.backup;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.ConsistencyLevel;
+import com.datastax.driver.core.KeyspaceMetadata;
+import com.datastax.driver.core.Metadata;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.Statement;
+import com.datastax.driver.core.TableMetadata;
+import com.datastax.driver.core.Token;
+import com.datastax.driver.core.TokenRange;
+import com.datastax.driver.core.querybuilder.QueryBuilder;
+import com.google.common.collect.Streams;
+import com.palantir.atlasdb.cassandra.CassandraKeyValueServiceConfig;
+import com.palantir.atlasdb.cassandra.CassandraServersConfigs.CqlCapableConfig;
+import com.palantir.atlasdb.cassandra.CassandraServersConfigs.DefaultConfig;
+import com.palantir.atlasdb.cassandra.CassandraServersConfigs.Visitor;
+import com.palantir.atlasdb.cassandra.proxy.SocksProxyNettyOptions;
+import com.palantir.atlasdb.keyvalue.cassandra.async.client.creation.ClusterFactory;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.logger.SafeLogger;
+import com.palantir.logsafe.logger.SafeLoggerFactory;
+import java.net.InetSocketAddress;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public final class CqlCluster {
+    private static final SafeLogger log = SafeLoggerFactory.get(CqlCluster.class);
+
+    // this is the partition key column name for the coord/TS tables
+    private static final String KEY_NAME = "key";
+    private static final int LONG_READ_TIMEOUT_MS = (int) TimeUnit.MINUTES.toMillis(2);
+    // reduce this from default because we run RepairTableTask across N keyspaces at the same time
+    private static final int SELECT_FETCH_SIZE = 1_000;
+
+    private final Cluster cluster;
+    private final CassandraKeyValueServiceConfig config;
+
+    private CqlCluster(Cluster cluster, CassandraKeyValueServiceConfig config) {
+        this.cluster = cluster;
+        this.config = config;
+    }
+
+    public static CqlCluster create(CassandraKeyValueServiceConfig config) {
+        // TODO(gs): error handling/retry
+        Set<InetSocketAddress> hosts = getHosts(config);
+        InetSocketAddress firstHost = hosts.stream().findAny().orElseThrow();
+        Cluster cluster = new ClusterFactory(
+                        () -> Cluster.builder().withNettyOptions(new SocksProxyNettyOptions(firstHost)))
+                .constructCluster(hosts, config);
+        return new CqlCluster(cluster, config);
+    }
+
+    private static Set<InetSocketAddress> getHosts(CassandraKeyValueServiceConfig config) {
+        return config.servers().accept(new Visitor<>() {
+            @Override
+            public Set<InetSocketAddress> visit(DefaultConfig defaultConfig) {
+                // TODO(gs): fail instead here? can we assume CQL support?
```

Comment:
we probably need to check this with a sample of production config if we wanted to assume CQL support. I'd be a bit concerned because we don't mandate this in the general case, but _maybe_ most people do have it

Reply:
so the internal code assumes CQL support (it creates a Cluster object and then does CQL queries (I copied this logic over to CqlCluster).